src/Validator/RomanNumberBasicValidator.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mathematicator\Numbers\Validator;
+
+
+use Nette\StaticClass;
+use Nette\Utils\Strings;
+
+/**
+ * Checks whether the number is valid in original ancient set. (integers (0), 1-3999 and fractions /12)
+ */
+final class RomanNumberBasicValidator
+{
+	use StaticClass;
+
+	/**
+	 * @param string $romanNumber
+	 * @param bool $allowZero
+	 * @return bool
+	 * @see https://stackoverflow.com/a/267405/1044198
+	 */
+	public static function validate(string $romanNumber, $allowZero = false)
+	{
+		if (strlen($romanNumber) === 0) {
+			return false;
+		} elseif ($allowZero && $romanNumber === 'N') {
+			return true;
+		}
+
+		return strlen($romanNumber) > 0 && (bool) preg_match('/^M{0,3}(CM|CD|D?C{0,3})(XC|XL|L?X{0,3})(IX|IV|V?I{0,3})$/', Strings::upper($romanNumber));
```

Comment:
I think the function should allow the insertion of some other shapes as well.

For example `VIIIIIIIII` is valid to, but not optimal.

Reply:
Could you please mention a source that it's valid?

Reply:
Source in Czech language (last implementation): https://php.vrana.cz/prevod-rimskych-cislic.php

> Kód kontroluje, jestli jsou číslice seřazené od nejvyšší po nejnižší a jestli řetězec neobsahuje nepovolený znak (což jsou nutné podmínky při zápisu římských číslic). Nekontroluje počet opakování jednotlivých číslic (např. IIIII = 5 nebo IXIX = 18 projde) ani kompaktnost zápisu (např. VIV = 9 projde, i když se dá zapsat jako IX = 9). **Podle článku na Wikipedii jsou ale takové zápisy možné**. Pokud bychom chtěli zkontrolovat, jestli je číslo v kanonické formě, tak by asi nejjednodušší bylo na výsledku zavolat toRoman a podívat se, jestli vrátilo náš vstup.

Reply:
Fixed